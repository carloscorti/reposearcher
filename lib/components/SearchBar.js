import React, { useState, useEffect } from 'react';
import debounce from 'lodash.debounce';

import { connect } from 'react-redux';
import * as actions from '../actions';

const SearchBar = (props) => {
  const [searchTerm, setSearchTerm] = useState('');

  useEffect(() => {
    doSearch();
  }, [searchTerm]);

  const doSearch = debounce(() => {
    props.storeSearchTerm(searchTerm);
  }, 300);

  const handleChange = (e) => {
    setSearchTerm(e.target.value);
  };

  return (
    <>
      <label>Search Bar</label>
      <input
        type="search"
        placeholder="Enter Search Term"
        value={searchTerm}
        onChange={handleChange}
      />
    </>
  );
};

export default connect(null, actions)(SearchBar);
