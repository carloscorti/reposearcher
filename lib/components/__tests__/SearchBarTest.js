import React from 'react';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';

import SearchBar from '../SearchBar';

import renderer from 'react-test-renderer';

const mockStore = configureStore([]);

describe('SearchBar', () => {
  const testProps = {
    repos: [{}, {}, {}],
  };

  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      storeSearchTerm: () => jest.fn(),
    });

    component = renderer.create(
      <Provider store={store}>
        <SearchBar repos={testProps.repos} />
      </Provider>
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
