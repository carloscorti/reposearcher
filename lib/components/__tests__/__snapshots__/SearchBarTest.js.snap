// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
Array [
  <nav>
    <ul
      className="repo-nav"
    >
      <li>
        Repositories 
        3
      </li>
    </ul>
  </nav>,
  <div
    className="search-bar-wrapper"
  >
    <input
      className="search-bar"
      onChange={[Function]}
      placeholder="Find a repository..."
      type="search"
      value=""
    />
  </div>,
]
`;
